### ISSUE TITLE
Unable to leave insert mode after pasting long string

### ISSUE THREAD

**Issue by flxai** [neovim/neovim#14120]
- `nvim --version`: v0.4.4 Release
- `vim -u DEFAULTS` (version: 8.2) behaves differently? Yes
- Operating system/version: Arch Linux
- Terminal name/version: urxvt v9.22
- `$TERM`: rxvt-unicode-256color

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
```

* Enter insert mode
  * e.g. by pressing <kbd>i</kbd>
* Paste the following string: `xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx`
  * e.g. using <kbd>Shift</kbd> + <kbd>Insert</kbd>
  * Note that if the string is shorter and a single `x` is missing everything works fine

```
# Alternative for shell-related problems:
# env -i TERM=ansi-256color "$(which nvim)"
```

* Not reproducible with `ansi-256color`

### Actual behaviour

The following text appears within the editor and exiting insert mode e.g. via <kbd>Ctrl</kbd> + <kbd>C</kbd> or <kbd>Esc</kbd> is impossible.

```
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
^[[201~
```

It appears to me that ANSI escape sequences are not interpreted correctly.

### Expected behaviour

The following should be pasted and it should be possible to leave insert mode.

```
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

I'm unsure whether this is really a bug in neovim and apologize if it is not, since I cannot reproduce it using `ansi-256color`.
I'd appreciate if you have hints for further debugging.

**Comment by kylepollina** [comment]
I've had similar issues with 0.4.4 - I can't reproduce the bug with this string however. Sometimes it just happens randomly to me, I haven't been able to find a rhyme or reason to it. I use iterm2 on MacOS.

I have found occasionally when stuck in insert mode, that pasting the string a second time will allow me to leave insert mode. Really strange

**Comment by erw7** [comment]
Thanks for the report. However, this is an issue that has already been fixed in #12004.

Ref #11699.

**Comment by flxai** [comment]
Okay, thanks erw7.
